iphone/Classes/TiUIImageViewProxy.m

```diff
@@ -47,10 +51,34 @@ -(void)propagateLoadEvent:(NSString *)stateString
         [self willChangeSize];
     }
 #endif
+	
     if ([self _hasListeners:@"load"]) {
         NSDictionary *event = [NSDictionary dictionaryWithObject:stateString forKey:@"state"];
         [self fireEvent:@"load" withObject:event];
     }
+#ifdef TI_USE_KROLL_THREAD
+    else {
+        // Why do we do this?
+        // When running on kroll thread this is being called before the events are added.
+        // So we try to propagate this after the load event is added.
```

Comment:
@hansemannn 
I tested with TI_USE_KROLL_THREAD enabled/disabled in defines.h. But I strongly suggest you to try it as well. Always good to have a different set of eyes!